hadoop-hdds/common/src/test/java/org/apache/hadoop/hdds/protocol/MockDatanodeDetails.java

```diff
@@ -118,11 +128,21 @@ public static DatanodeDetails randomLocalDatanodeDetails()
       return createDatanodeDetails(UUID.randomUUID().toString(),
           socket.getInetAddress().getHostName(),
           socket.getInetAddress().getHostAddress(), null,
-          socket.getLocalPort());
+          getFreePort());
     }
   }
 
   private MockDatanodeDetails() {
     throw new UnsupportedOperationException("no instances");
   }
+
+  static synchronized int getFreePort() {
+    LOG.info("### Getting free port ###");
+    int port = NEXT_PORT.getAndIncrement();
+    if (port > MAX_PORT) {
+      NEXT_PORT.set(MIN_PORT);
+      port = NEXT_PORT.getAndIncrement();
+    }
+    return port;
+  }
```

Comment:
Please don't duplicate code unless absolutely necessary.

Move `PortAllocator` from `MiniOzoneCluster` to a top-level class in `hadoop-hdds/test-utils/src/main/java/org/apache/ozone/test`, so that it can be reused.

Its methods need to be exposed as `public`.  Also, please make the top-level class `final` (also required by checkstyle for utility classes).

Reply:
Thanks for the idea!
Have made the changes.